### PR TITLE
Convert the repository into a galaxy role

### PR DESCRIPTION
**Pull request by brtkwr**
Addition work to support use of bastion host to provision k8s cluster via AWX has been moved to https://github.com/cedadev/jasmin-appliances/pull/15

### CODE REVIEW

File: kubeadm/vars/external.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
```
Comment (mkjpryor-stfc) [inline kubeadm/vars/external.yml]: This file can be removed - it makes no sense without some additional work.

**Review by mkjpryor-stfc — CHANGES_REQUESTED** [review]

File: kubeadm/defaults/main.yml [hunk 2]
```diff
@@ -34,8 +34,14 @@ cluster_config_extra: {}
...
+openstack_project_domain_name: "{{ lookup('env', 'OS_PROJECT_DOMAIN_NAME') }}"
+openstack_domain_name: "{{ lookup('env', 'OS_DOMAIN_NAME') | default(openstack_project_domain_name) }}"
 openstack_project_id: "{{ lookup('env', 'OS_PROJECT_ID') }}"
+openstack_user_id: "{{ lookup('env', 'OS_USER_ID') }}"
+openstack_trust_id: "{{ lookup('env', 'OS_TRUST_ID') }}"
+openstack_project_name: "{{ lookup('env', 'OS_PROJECT_NAME') }}"
 openstack_username: "{{ lookup('env', 'OS_USERNAME') }}"
 openstack_password: "{{ lookup('env', 'OS_PASSWORD') }}"
+openstack_token: "{{ lookup('env', 'OS_TOKEN') }}"
+openstack_auth_type: "{{ lookup('env', 'OS_AUTH_TYPE') }}"
```
Comment (mkjpryor-stfc) [inline kubeadm/defaults/main.yml]: Just one more change. We should only have variables in here that are actually used by the `kubeadm` role. That looks like only `openstack_auth_url`, `openstack_trust_id`, `openstack_trustee_id` and `openstack_trustee_password`.

**Review by mkjpryor-stfc — CHANGES_REQUESTED** [review]

File: kubeadm/defaults/main.yml (hunk 2 above)
Comment (mkjpryor-stfc) [reply kubeadm/defaults/main.yml]: `openstack_project_id` is also used by the authentication.

Comment (brtkwr) [reply kubeadm/defaults/main.yml]: Done

**Review by mkjpryor-stfc — APPROVED** [review]

**Pull request merged**